fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ReplaceColumnRefRewriter.java

```diff
@@ -45,6 +45,10 @@ public ScalarOperator visitVariableReference(ColumnRefOperator column, Void cont
         // Rewiring predicate shouldn't change the origin project columnRefMap
 
         ScalarOperator mapperOperator = operatorMap.get(column).clone();
+        // Keep the original attribute of nullable
+        if (mapperOperator.isColumnRef()) {
```

Comment:
This should be the clone method wrong.  we should keep nullable info when clone.

Reply:
Column is the key, clone performed on value. I think we cannot change some attributes(isNullable is one of them) when replacing columnRef with scalarOperator.